src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
+   buffer = nn_allocmsg(data_size, 0);
+   auto error = nn_errno ();
+   if (error == ENOMEM) {
+      throw std::runtime_error("not enough memory to allocate NanoMsg");
```

Comment:
we probably should talk about it, i view the out of memory condition as an extremely bad situation, to me it makes sense to bring down the app if memory is finished as if it keeps running it will exacerbate the condition. if i recall correctly OOM errors don't start until you exhaust swap  also.

in general i'm not a fan of handling an exception if it's not your responsibility, for example if this class doesn't have the memory to fulfill it's purpose then it's up to who ever is using it to handle it if they want. persisting the class when it cannot serve its purpose is weird to me.